### PR TITLE
8340809: Open source few more AWT PopupMenu tests

### PR DESCRIPTION
**Pull request by kumarabhi006**
Few more AWT PopupMenu tests converted to either manual or automated and open sourced.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8340809](https://bugs.openjdk.org/browse/JDK-8340809): Open source few more AWT PopupMenu tests (**Bug** - P4)

### Reviewers
 * [Alexey Ivanov](https://openjdk.org/census#aivanov) (@aivanov-jdk - **Reviewer**) Review applies to [96975b10](https://git.openjdk.org/jdk/pull/21412/files/96975b10c91e8b10f047620e8c155312d5a4509e)
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21412/head:pull/21412` \
`$ git checkout pull/21412`

Update a local copy of the PR: \
`$ git checkout pull/21412` \
`$ git pull https://git.openjdk.org/jdk.git pull/21412/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21412`

View PR using the GUI difftool: \
`$ git pr show -t 21412`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21412.diff">https://git.openjdk.org/jdk/pull/21412.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21412#issuecomment-2400406977)

### CODE REVIEW

File: test/jdk/java/awt/PopupMenu/ActivePopupCrashTest.java [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+        pm.add(mi);
+
+        mi = new MenuItem(name + "-2");
+        pm.add(mi);
+
+        setHash(c, pm);
+        c.add(pm);
+        c.addMouseListener(new MouseAdapter() {
+            @Override
+            public void mouseClicked (MouseEvent e) {
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/PopupMenu/ActivePopupCrashTest.java:158]:
```suggestion
            public void mouseClicked(MouseEvent e) {
```
No space.

File: test/jdk/java/awt/PopupMenu/ActivePopupCrashTest.java [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+                mouseAction("mouseClicked", e);
+            }
+
+            @Override
+            public void mousePressed (MouseEvent e) {
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/PopupMenu/ActivePopupCrashTest.java:163]:
```suggestion
            public void mousePressed(MouseEvent e) {
```

File: test/jdk/java/awt/PopupMenu/ActivePopupCrashTest.java [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+                mouseAction("mousePressed", e);
+            }
+
+            @Override
+            public void mouseReleased (MouseEvent e) {
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/PopupMenu/ActivePopupCrashTest.java:168]:
```suggestion
            public void mouseReleased(MouseEvent e) {
```

File: test/jdk/java/awt/PopupMenu/MultiplePopupMenusTest.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+import java.awt.Frame;
+import java.awt.MenuItem;
+import java.awt.PopupMenu;
+
+import java.awt.event.MouseEvent;
+
+/*
+ * @test
+ * @bug 4186663 4265525
+ * @summary Tests that plural PopupMenus can not appear at the same time
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/PopupMenu/MultiplePopupMenusTest.java:35]:
```suggestion
 * @summary Tests that multiple PopupMenus cannot appear at the same time
```

File: test/jdk/java/awt/PopupMenu/PopupMenuCrash.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+ * @build PassFailJFrame
+ * @requires (os.family == "windows")
+ * @run main/manual PopupMenuCrash
+ */
+
+public class PopupMenuCrash {
+    public static void main(String[] args) throws Exception {
+        String INSTRUCTIONS = """
+                This tests a windows specific problem.
+                When you see a frame titled "PopupMenuCrash Test", right-click on it a lot
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/PopupMenu/PopupMenuCrash.java:48]:
```suggestion
                When you see a frame titled "PopupMenuCrash Test", right-click on it several times
```
Does it sound better?

File: test/jdk/java/awt/PopupMenu/PopupMenuCrash.java [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+        f.addMouseListener(new MouseAdapter() {
+            PopupMenu popup;
+            boolean firstPress = true;
+
+            @Override
+            public void mousePressed(MouseEvent evt) {
+                if (firstPress) {
+                    firstPress = false;
+                    try {
+                        Thread.sleep(15000);
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/PopupMenu/PopupMenuCrash.java:78]:
```suggestion
                        Thread.sleep(10000);
```
Reduce to 10 seconds as stated in instructions?

**Review by aivanov-jdk — APPROVED** [review]
Looks good to me overall. There are a few nits.

File: test/jdk/java/awt/PopupMenu/PopupMenuCrash.java (hunk 6 above)
Comment (kumarabhi006) [reply test/jdk/java/awt/PopupMenu/PopupMenuCrash.java]: Updated.

File: test/jdk/java/awt/PopupMenu/PopupMenuCrash.java (hunk 5 above)
Comment (kumarabhi006) [reply test/jdk/java/awt/PopupMenu/PopupMenuCrash.java]: Yes... Updated.

**Pull request closed without merging**